riscv/execute.cc

```diff
@@ -14,7 +14,7 @@ static void commit_log_reset(processor_t* p)
   p->get_state()->log_mem_write.clear();
 }
 
-static void commit_log_stash_privilege(processor_t* p)
+static void stash_privilege(processor_t* p)
```

Comment:
Why change the function name? It clutters up this commit with other things, obscuring the substantive changes. If it really makes sense to change the function name, that should be a separate commit.

Reply:
Please see new commit to see if it better explains the rationale.

Reply:
The stashed state information saves the previous state before the execution of the instruction. The previous implementation uses the information only for logging after the execution and, thus, maintains the variables only when logging is enabled.

This PR points out the stashed state information is also required for the desired functionality, i.e., state.last_inst_priv for trigger checking. Thus, the PR proposes maintaining the information unconditionally regardless of whether in fast/slow path or logging is enabled or not. (Otherwise, we may need to introduce another variable with the same value if valid. That was the previous proposal of this PR. https://github.com/riscv-software-src/riscv-isa-sim/pull/1350#discussion_r1191791795)

Also, changing variable validity by modifying the calling condition and renaming the function for better description could be two rational independent commits, which may provide more clear reasoning for this PR.

Reply:
@YenHaoChen Good suggestion; updated per the above.